x/evm/types/config_testing.go

```diff
@@ -58,8 +58,8 @@ func (ec *EVMConfigurator) WithChainConfig(cc *ChainConfig) *EVMConfigurator {
 }
 
 // WithEVMCoinInfo allows you to define the denom and decimals of the token used as the EVM token.
-func (ec *EVMConfigurator) WithEVMCoinInfo(denom string, d Decimals) *EVMConfigurator {
-	ec.evmDenom = EvmCoinInfo{Denom: denom, Decimals: d}
+func (ec *EVMConfigurator) WithEVMCoinInfo(denom string, d int64) *EVMConfigurator {
```

Comment:
also the current `decimals` type is rather `uint64` which is causing a lot of complaints for the linter with `int64 -> uint64` conversions

Reply:
Hey Malte, thank for the review! I was trying to create a config consistent with the registration of unit of the denom which is using LegacyDecimals:
https://github.com/evmos/evmos/blob/f7e49cf57b864d8a4143dd5ec88dfa1ca1492551/app/config.go#L85-L85

But It was hard to retrieve the exponent of the representation given the number (required an inversion of the the unit stored). I believe in the next iteration we should write our custom `RegisterDenom`, since it is will also be deprecated in future Cosmos SDK. I've already restored previous definition using uint32, but as you mention we could also use uint8.